ivy_tests/test_ivy/helpers/hypothesis_helpers/array_helpers.py

```diff
@@ -780,6 +780,22 @@ def array_values(
                         exclude_max=exclude_max,
                     ),
                 )
+                if "float" in dtype or "complex" in dtype:
+                    kind_dtype = "float"
+                    dtype_info = ivy.finfo(dtype)
+                elif "int" in dtype:
+                    kind_dtype = "int"
+                    dtype_info = ivy.iinfo(dtype)
+                elif "bool" in dtype:
+                    # new code to generate random float array with subnormal values
+                    if allow_subnormal:
```

Comment:
This logic doesn't seem intuitive to me, can you explain further why is this done in `array_helpers`? we shouldn't be generally assuming that kind of input to produce a specific case for the function we're trying to fix, an input of type `bool` should **ALWAYS** produce an array of booleans and ignore any other arguments related to other data types. 

I'd suggest solving that specific case by using a composite strategy instead.

Reply:
Okay, I get, if you go through my comments I have explaimed why I  had it in array_helpers. Yes, I understand the concern about assuming that a specific data type should produce a specific case for the function. It's always better to have a more flexible and composite strategy that takes into account multiple scenarios and inputs. Let me correct that